ucsm_apis/network/vlan.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2017 Cisco Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module performs the operation related to user.
+"""
+from ucsmsdk.ucsexception import UcsOperationError
+_base_dn = "fabric/lan"
+
+
+def vlan_create(handle, name, id, sharing=None, mcast_policy_name=None, policy_owner=None,
+                default_net=None, compression_type=None, **kwargs):
+    """
+        creates vlan
+        Args:
+            handle (UcsHandle)
+            name (string): VLAN name
+            sharing (string): sharing, valid values "community", "isolated", "none", "primary"
+            mcast_policy_name (string):
+            policy_owner (string): "local", "pending-policy", "policy"
+            default_net (string): "false", "no", "true", "yes"
+            compression_type (string): "excluded", "included"
+            **kwargs: Any additional key-value pair of managed object(MO)'s
+                      property and value, which are not part of regular args.
+                      This should be used for future version compatibility.
+        Returns:
+            FabricVlan: managed object
+        Raises:
+            None
+        Example:
+            vlan_create(handle, name="test", id=2, sharing="none", mcast_policy_name="", policy_owner="local",
+                        default_net="no", compression_type="included")
+        """
+    from ucsmsdk.mometa.fabric.FabricVlan import FabricVlan
+    vlan = FabricVlan(parent_mo_or_dn=_base_dn, name=name, sharing=sharing, id=id,
```

Comment:
@ReignInChaos this is perfect for global vlan, we need to see how to use the same APIs for other vlans. May be we could provide the `base_dn` in vlan_create and default it to `fabric/lan`. that way, fabric specific vlans would also just work by providing a different `base_dn` argument value. This would need similar changes in `vlan_get` etc..